experiments/scan_classes/averaging_periods.py

```diff
@@ -0,0 +1,143 @@
+        # make a list of the cpos in this AveragingPeriod.
+
+        ScanClassBase.__init__(self, ave_keys, ave_slice_dict, ave_interface, options)
+
+        self.slice_to_beamorder = slice_to_beamorder_dict
+        self.slice_to_beamdir = slice_to_beamdir_dict
+
+        # Metadata for an AveragingPeriod: clear frequency search, integration time, number of averages goal
+        self.clrfrqf = False
+        # there may be multiple slices in this averaging period at different frequencies so
+        # we may have to search multiple ranges.
+        self.clrfrqrange = []
+        for slice_id in self.slice_ids:
+            if self.slice_dict[slice_id]['clrfrqflag']:
+                self.clrfrqf = True
+                self.clrfrqrange.append(self.slice_dict[slice_id]['clrfrqrange'])
+
+        # TODO: SET UP CLEAR FREQUENCY SEARCH CAPABILITY
+        # also note for when setting this up clrfrqranges may overlap
+
+        self.intt = self.slice_dict[self.slice_ids[0]]['intt']
+        self.intn = self.slice_dict[self.slice_ids[0]]['intn']
+        if self.intt is not None:  # intt has priority over intn
+            for slice_id in self.slice_ids:
+                if self.slice_dict[slice_id]['intt'] != self.intt:
+                    errmsg = """Slice {} and {} are INTTIME mixed and do not have the
+                        same Averaging Period duration intt""".format(self.slice_ids[0], slice_id)
+                    raise ExperimentException(errmsg)
+        elif self.intn is not None:
+            for slice_id in self.slice_ids:
+                if self.slice_dict[slice_id]['intn'] != self.intn:
+                    errmsg = """Slice {} and {} are INTTIME mixed and do not have the
+                        same NAVE goal intn""".format(self.slice_ids[0], slice_id)
+                    raise ExperimentException(errmsg)
+
+        for slice_id in self.slice_ids:
+                if len(self.slice_dict[slice_id]['beam_order']) != len(self.slice_dict[self.slice_ids[0]]['beam_order']):
+                    errmsg = """Slice {} and {} are INTTIME mixed and do not have the
+                        same length of beam_order (number of integration periods)
+                        """.format(self.slice_ids[0], slice_id)
+                    raise ExperimentException(errmsg)
+
+        # NOTE: Do not need beam information inside the AveragingPeriod, this is in Scan.
+
+        # Determine how this averaging period is made by separating out
+        #   the INTEGRATION mixed.
+
+        self.slice_sequence_list = self.get_sequences()
+        self.sequences = []
+        self.nested_slice_list = self.slice_sequence_list
+
+        for params in self.prep_for_nested_scan_class():
+            self.sequences.append(Sequence(*params))
+
+    def get_sequences(self):
+        integ_combos = []
+
+        # Remove INTEGRATION combos as we are trying to separate those.
+        for k in self.interface.keys():
+            if self.interface[k] == "PULSE":
```

Comment:
No. self.interface in the AveragingPeriod class will have only the keys INTEGRATION and PULSE, but here we are trying to pull out only those that are PULSE interfaced so that we can make sequences with those ones. I'll make an example to document.